net/interfaces/interfaces_darwin.go

```diff
@@ -62,6 +63,9 @@ func likelyHomeRouterIPDarwinExec() (ret netaddr.IP, ok bool) {
 		ip, err := netaddr.ParseIP(string(mem.Append(nil, ipm)))
 		if err == nil && isPrivateIP(ip) {
 			ret = ip
+			// We've found what we're looking for.
+			// Return any error to stop reading.
+			return errors.New("success")
```

Comment:
this allocates just to stop it.

use some global?

Reply:
It's a single alloc in infrequently run code that is only used when cgo is disabled. I've switched to a global, but it really doesn't matter.